### DISCUSSION TITLE
Question about result of the molfile_to_params.py

### DISCUSSION THREAD

**Discussion by kn301BE** [RosettaCommons/rosetta#177] [Rosetta] [answered]
Hi! I was trying to use the script molfile_to_params.py to get the params file for a very short peptide which only contains several residues but the script always recognized te peptide as a chemical ligand. I found there was a script called molfile_to_params_polymer_main.py which make write a polymer param file instead of ligand param file but it seem to be not found in the rosetta 3.13 version and there is no argument which could do this in the current script. Would someone could help me about how to get a params file and the pdb file for a peptide instead of a ligand params file and ligand pdb file.

**Comment by roccomoretti** [comment, marked as answer]
Rosetta works primarily at the residue level. You need to have parameters for each "residue". For ligand docking and the like, we treat the entire small molecule as a single "residue" -- so the molfile_to_params.py script is set up to take the input and make it a single (non-polymeric) "residue".

If you're looking at a polymer (including peptides), you don't need to parameterize the polymer as a whole, instead each individual monomeric unit in the polymer is parameterized. Rosetta comes with a number of pre-parameterized residue types -- this is why it can read in standard proteins. In the Rosetta database there's parameter files for each of the 20 canonical amino acids and "patch files" for many of the standard modifications, so when Rosetta reads in a standard protein, it knows how to handle each residue it encounters.

If your peptide contains only canonical amino acids (i.e. the 20 standard ones) and you want to treat it as a polymeric chain, then you don't have to make parameters for it -- simply include it in your input PDB file as a collection of residues ... in the same format which you would get it from the wwPDB.

You only have to worry about parameterization (e.g. molfile_to_params_polymer) if you have non-canonical amino acids. But even then, chances are that if you have a "common" non-canonical amino acid there's already a parameterize of that NCAA in the database. All you have to do is to figure out what the three letter code Rosetta uses for that NCAA and make sure your input file matches with that three letter code.  (Note though, that even if the NCAA is in the database, the rotamer library for that residue might be distributed separately - see the downloads page at rosettacommons.org to get the files.)

If you do have an NCAA residue which doesn't currently have a parameter file in the database, you will need to generate one with molfile_to_params -- There's been some updates to the process recently, which may be why you can't find the script where you expect in Rosetta 3.13. If you do need to go down this route, I'd recommend updating to Rosetta 3.14, and taking a look at the [Macrocyclic peptide design with Noncanonical Amino Acids](https://meilerlab.org/rosetta-workshop-2024/) tutorial from the Meiler Lab.
